### PR TITLE
add tc api

### PR DESCRIPTION
**Pull request by yanivagman**
This PR adds the low level tc api as described in https://github.com/libbpf/libbpf/commit/d71ff87a2dd7b92787719aab233767e9c74fbd48

### CODE REVIEW

File: libbpfgo.go [hunk 1]
```diff
@@ -1038,3 +1039,143 @@ func (pb *PerfBuffer) poll() error {
...
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return err
+	}
+	hook.hook.ifindex = C.int(iface.Index)
+
+	return nil
+}
+
+func (hook *TcHook) GetInterfaceIndex() int {
```
Comment (grantseltzer) [inline libbpfgo.go:1119]: What would users need this for?

**Review by grantseltzer — COMMENTED** [review]
Do I understand correctly that this allows users to attach bpf programs to tc actions, where tc actions are when tc filters are modified in some ways? 

I.e.) User adds a tc filter for slowing down egress on a particular interface. The bpf program would have access to information about that action.

File: libbpfgo.go (hunk 1 above)
Comment (yanivagman) [reply libbpfgo.go:1119]: Not sure that this would be needed, but I thought it may be good to have. I don't mind removing it though.

**Comment by yanivagman** [comment]
> Do I understand correctly that this allows users to attach bpf programs to tc actions, where tc actions are when tc filters are modified in some ways?
> 
> I.e.) User adds a tc filter for slowing down egress on a particular interface. The bpf program would have access to information about that action.

**Review by grantseltzer — APPROVED** [review]
LGTM!

**Pull request merged**